intelmq/bots/experts/filter/expert.py

```diff
@@ -1,22 +1,79 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
+
+import re
 import sys
+from datetime import datetime, timedelta
 
+from dateutil import parser
 from intelmq.lib.bot import Bot
+from intelmq.lib.harmonization import DateTime
+
 
+class FilterExpertBot(Bot):
 
-class FilterBot(Bot):
+    # number of minutes in time units
+    timespans = ({'hour': 60, 'day': 24*60, 'week': 7*24*60, 'month': 30*24*60, 'year': 365*24*60})
+
+    # parse relative time attributes
+    @staticmethod
+    def parse_relative(relative_time):
+        try:
+            result = re.findall(r'(\d+)\s+(\w+)?s', relative_time, re.UNICODE)
+        except Exception as e:
+            raise AttributeError("Could not apply regex to attribute \"%s\" with exception %s",
+                                 repr(relative_time), repr(e.args))
+        if len(result) == 1 and len(result[0]) == 2 and result[0][1] in FilterExpertBot.timespans:
+            return int(result[0][0]) * FilterExpertBot.timespans[result[0][1]]
+        else:
+            raise AttributeError("Could not process result of regex for attribute " + repr(relative_time))
+
+    # decide format of timefilter value and parse it
+    def parse_timeattr(self, time_attr):
+        try:
+            absolute = parser.parse(time_attr)
+        except:
```

Comment:
As already said, we don't want to catch all Exceptions here (no IOError, MemoryError, OverflowError, SystemError, SystemExit etc.)